ocaml/xapi/xapi_vdi.ml

```diff
@@ -128,6 +152,9 @@ let check_operation_error ~__context ?(sr_records=[]) ?(pbd_records=[]) ?(vbd_re
       then Some (Api_errors.vdi_in_use,[_ref; (Record_util.vdi_operation_to_string op)])
       else if my_has_current_operation_vbd_records <> []
       then Some (Api_errors.other_operation_in_progress, [ "VDI"; _ref ])
+      else let sm_feature_error = check_sm_feature_error op sm_features sr in
```

Comment:
Actually, we could also move obtaining the `sm_features` into the `check_sm_feature_error` function, but that would require an extra context parameter though.